### DISCUSSION TITLE
Dynamic pipeline

### DISCUSSION THREAD

**Discussion by frant1** [savonet/liquidsoap#3771] [Q&A] [answered]
Hello !

I'm new to Liquidsoap, I used it though Airtime and Azuracast but now I need to put my hands in it so I'm creating my own scripts.

Here is the context : I have a webradio portal, each channel output an icecast stream after the audio processing, the icecast is currently processed by a tool and then transformed in a SRT stream.. Currently the transcoding tool isn't optimized for icecast and I think that we could make it with Liquidsoap, so I made it for a channel and it works (Only on Linux because of the #2461 issue). I would liked to host the script on Windows because that would have made things easy.. (each channel is lonely VM instances).

In reason of this problem I have to host it using a Linux farm. Now the fact is the pipelines are not statics, there are new almost every months and it's going to be a problem to need to restart every streams to add one..

I saw the doc and I thought that we could use the provided telnet or feature to dynamically add, list or delete a stream. The fact is I wasn't able to make it. I have some experiences with programming language (Golang / Python / Swift..) but I'm still having difficulties with the Liquidsoap's one.
Precision : I'm trying to make a farm in a single Liquidsoap process because I saw that making 50 Liquidsoaps daemons is using much more perfs than making one Liquidsoap daemon transcoding 50 streams..

Does an expert of Liquidsoap would have an idea about the ability of Liquidsoap to do it ?
Does someone would already have met a similar case ?

Best regards !

**Comment by vitoyucepi** [comment, marked as answer]
Hi @frant1,

I think it is possible to create a dynamic pipeline, but I haven't seen anyone do it.
For example, the *Dynamic sources* section on page 207 of [the book](https://www.liquidsoap.info/book/book.pdf) explains how to create and control sources after the process has started.
It's even possible to create a visual pipeline editor, but that would require some serious development outside of liquidsoap.
You could use http+json to transfer your payload to and from the liquidsoap.

> I saw that making 50 Liquidsoaps daemons is using much more perfs than making one Liquidsoap daemon transcoding 50 streams.

The common practice used to be to give each user a separate process to avoid mixing businesses.
